src/Smalot/PdfParser/Page.php

```diff
@@ -54,6 +54,13 @@ class Page extends PDFObject
      */
     protected $dataTm;
 
+    public function setFonts($fonts)
+    {
+        if (empty($this->fonts)) {
+            $this->fonts = $fonts;
```

Comment:
This method looks weird. Why overriding `$this->fonts` only if it is empty?

Reply:
That's a protection in case the Page object already has individual fonts attached from the document.  This way it only gets fonts from the Pages object if it has none of its own.